orttraining/orttraining/python/training/ortmodule/_io.py

```diff
@@ -306,23 +316,29 @@ def _extract_schema(data):
     elif isinstance(data, torch.Tensor):
         return _TensorStub(dtype=str(data.dtype), shape_dims=len(data.size()))
 
+    # Instead of replacing the tensor with a stub in the original user input, build the stubbed_schema
+    # from scratch from the user input.
```

Comment:
out of curiosity, why this approach change? reusing the list saves memory, as resize of list is not needed. by building a new one, the list will exceeds its size from time to time and a full copy happens

Reply:
what was happening was that if the user input a a mutable type (lets say a dictionary), we were replacing the tensor value with a tensorstub in the original user dictionary. As a result, the user input dictionary was now updated (along with our cached schema). It was either this or do a deepcopy of the user input (which would be more inefficient).